### PR TITLE
[TASK] allow 'hide default translation of page' on root page

### PR DESCRIPTION
**Pull request by maritwho**
If the root page is set to 'hide default translation of page' this causes a 404 in TypoScriptFrontendController with initializing the TSFE. To avoid this a language uid has to be thrown to the initialize function.

Fixes: #1374

### CODE REVIEW

File: Classes/IndexQueue/Indexer.php [hunk 1]
```diff
@@ -554,7 +555,17 @@ protected function getSolrConnectionsByItem(Item $item)
             $siteLanguages[] = $solrConfiguration['language'];
         }
 
-        $translationOverlays = $this->getTranslationOverlaysForPage($pageId, $site->getSysLanguageMode());
+        $defaultLanguageUid = 0;
```
Comment (timohund) [inline Classes/IndexQueue/Indexer.php]: I would propose to move this into a method getDefaultLanguageUid()

Comment (maritwho) [inline Classes/IndexQueue/Indexer.php]: Ok, I will change this and update my pull request.

**Comment by maritwho** [comment]
So now I updated my pull request. There is another change in line 571 I forgot in my first push. I implemented now a function getDefaultLanguageUid.

I will try to add a test for this in the next view days.

**Comment by timohund** [comment]
Thanks for the optimizations of the patch. I think when we have a test for the use case we can merge it and should also port the fix to dev-master as well.

**Comment by timohund** [comment]
@maritwho Are you still working on this? For me it looks like that your fixture needs to be located in  Tests/Integration/IndexQueue/Fixtures, not  Tests/Integration/Fixtures ?

**Comment by maritwho** [comment]
@timohund Yes I am still working on this. I didn't have enough time to fix this. Thank you for your hint!

**Comment by timohund** [comment]
@maritwho is there any update on this? Do you think you will have the time to continue on that?

**Comment by maritwho** [comment]
I tried something in a separate branch on  my fork and travis. I was really busy the last weeks, so I had no time. But may be we can have a short talk at the T3DD17 in Malmö, so that I can finish this?

**Comment by timohund** [comment]
@maritwho sure good idea, maybe in the coding night?

**Comment by maritwho** [comment]
@timohund This pull request can be deleted. It's done in #1409

**Pull request closed without merging**